hermes_nemisis/calibration/calibration.py

```diff
@@ -155,11 +149,31 @@ def parse_l0_sci_packets(data_filename: Path) -> dict:
     pkt = ccsdspy.FixedLength.from_file(
         Path(hermes_nemisis._data_directory) / "NEM_sci_packet_def.csv"
     )
+
+    if (data_filename.stat().st_size == 0):
+        raise ValueError(f"Zero length file: {data_filename}")
     data = pkt.load(data_filename)
+
+    # calculate checksum of each packet
+    n_pkts = len(data['CHECKSUM'])
+    packet_bytes = ccsdspy.utils.split_packet_bytes(
+        data_filename,
+        include_primary_header=False
+    )
+    calculated_checksum_out = np.zeros(n_pkts, dtype=np.uint16)
+    for i in range(n_pkts):
+        CK_A = 0 
+        CK_B = 0
+        for octet_num in range(1115):
```

Comment:
Is this always going to be the same? Is this 1115 a constant or is it potentially variable? If its a constant, maybe ad a comment on where it comes from.

Reply:
Thank you for adding this comment. This makes much more sense.